### PR TITLE
docs: add SECURITY.md, security reporting process

### PR DESCRIPTION
**Pull request by dorianj**
### Summary of Changes

This documents the new security vulnerability mailing list.

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely, including a title prefix.
- [ ] PR includes a summary of changes.
- [ ] My commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+We don't currently have a PGP key, unfortunately.
+
+An Amundsen committer will send you a response indicating the next steps in handling your report. After the initial reply to your report, the committer will keep you informed of the progress towards a fix and full announcement, and may ask for additional information or guidance.
+
+*Important:* Please don't disclose the vulnerability before it have been fixed and announced, to protect our users.
+
+## Security announcements
+
+Please subscribe to the main mailing list, where we post notifications and remediation details for security vulnerabilities.
```
Comment (Golodhros) [inline SECURITY.md:14]: Would be useful to add a link to that mailing list, @markgrover do you have it handy?

**Review by Golodhros — APPROVED** [review]
LGTM

I would add a link from the main Readme to this document.

File: SECURITY.md (hunk 1 above)
Comment (feng-tao) [reply SECURITY.md:14]: there is an email link in the above (amundsen-security@lists.lfaidata.foundation). I ask the question in the channel as well, but do we know LF have any established handling security issue? or it relies on each project to handle on its own.

Comment (dorianj) [reply SECURITY.md:14]: Pasting from Slack, so it's visible publicly:
```
LF recommends:
Identify who on the project will be handling security issues (could be a team) 
Set up a separate email account to receive reporting (publish that on GH and project web site) 
Consider having the project become a CNA (CVE Numbering Authority)
```

Comment (feng-tao) [reply SECURITY.md:14]: thanks

**Review by feng-tao — APPROVED** [review]

File: SECURITY.md (hunk 1 above)
Comment (Golodhros) [reply SECURITY.md:14]: Then this is not a "main" mailing list by the "security" mailing list, right?

Comment (dorianj) [reply SECURITY.md:14]: it's this https://lists.lfai.foundation/g/amundsen-announce , adding it to the PR

**Comment by dorianj** [comment]
Updated to add link to announcements mailing list, and added to main README. Can someone with write bits please merge? Thank you

**Review by Golodhros — APPROVED** [review]
LGTM

Thanks for driving this Dorian!

**Pull request merged**